src/geo/transform.js

```diff
@@ -305,6 +326,109 @@ class Transform {
+                yCenter - r * yRadius * Math.cos(theta)
+            ]);
+        }*/
+
+        /*
+        // BEGIN DEBUG
+        const DEBUG = true;
+        let debugEl;
+        if (DEBUG) {
+            debugEl = document.getElementById('pts');
+            if (!debugEl) {
+                debugEl = document.createElement('div');
+                debugEl.id = 'pts';
+                debugEl.style.position = 'absolute';
+                debugEl.style.top = 0;
+                debugEl.style.bottom = 0;
+                debugEl.style.left = 0;
+                debugEl.style.right = 0;
+                debugEl.style.zIndex = 10;
+                debugEl.style.pointerEvents = 'none';
+                document.body.appendChild(debugEl);
+
+                for (let i = 0; i < points.length; i++) {
+                    const pt = document.createElement('span');
+                    pt.style.position = 'absolute';
+                    pt.style.width = '10px'
+                    pt.style.height = '10px'
+                    pt.style.borderRadius = '10px';
+                    pt.style.backgroundColor = 'red';
+                    pt.style.transform = 'translate(-5px,-5px)';
+                    pt.style.textIndent = '1em';
+                    pt.style.textShadow = '0 0 2px #ffffff, 0 0 2px #ffffff, 0 0 2px #ffffff, 0 0 2px #ffffff, 0 0 2px #ffffff';
+                    debugEl.appendChild(pt);
+                }
+            }
+
+            for (let i = 0; i < points.length; i++) {
+                const el = debugEl.children.item(i);
+                el.style.left = `${points[i][0]}px`;
+                el.style.top = `${points[i][1]}px`;
+            }
+        }
+        // END DEBUG
+        */
+
+        let elevationSum = 0.0;
+        let weightSum = 0.0;
+        if (this._elevation) {
+            for (let i = 0; i < points.length; i++) {
+                const p = points[i];
+                const hit = this._elevation.pointCoordinate(new Point(p[0], p[1]));
+                if (!hit) {
+                    /*
+                    // BEGIN DEBUG
+                    debugEl.children.item(i).textContent = 'null'
+                    debugEl.children.item(i).style.backgroundColor = 'gray';
+                    // END DEBUG
+                    */
+                    continue;
+                }
+                /*
+                // BEGIN DEBUG
+                if (DEBUG) {
+                    debugEl.children.item(i).textContent = hit[3].toFixed(0);
+                    debugEl.children.item(i).style.backgroundColor = 'blue';
+                }
+                // END DEBUG
+                */
+
+                const weight = 1 / Math.hypot(hit[0] - this._camera.position[0], hit[1] - this._camera.position[1]);
+                elevationSum += hit[3] * weight;
+                weightSum += weight;
+            }
+        }
+
+        return elevationSum / Math.max(weightSum, 1e-8);
```

Comment:
1e-8 -> Number.EPSILON, unless this one should also match AVERAGE_ELEVATION_CHANGE_THRESHOLD

Reply:
I've replaced it with a conditional. The non-zeroness is irrelevant; it's just a fix when no samples were taken and weight ==== 0.